modules/investigating-kernel-crashes.adoc

```diff
@@ -0,0 +1,95 @@
+// Module included in the following assemblies:
+//
+// * support/troubleshooting/troubleshooting-operating-system-issues.adoc
+
+[id="investigating-kernel-crashes"]
+= Investigating kernel crashes
+
+== Enabling kdump
+
+`kdump` (included in `kexec-tools`) is a service providing a crash dumping mechanism. The service enables you to save the contents of the system’s memory for later analysis.
+
+{op-system} ships with `kexec-tools`, but manual configuration is required to enable `kdump`.
+
+The following are the basic steps needed to enable `kdump` on {op-system}.
+
+.  Memory must be reserved for the crash kernel during booting of the first kernel. Kernel arguments can be provided like this:
++
+[source, terminal]
+----
+sudo rpm-ostree kargs --append='crashkernel=256M'
+----
+
+.  By default, the path in which the vmcore will be saved is `/var/crash`. It is also possible to write the dump over the network or to some other location on the local system by editing `/etc/kdump.conf`. For additional information, see `kdump.conf(5)` and the comments in `/etc/kdump.conf` and `/etc/sysconfig/kdump`.
+ifdef::openshift-enterprise[]
+Also refer to the RHEL `kdump` documentation link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/system_design_guide/installing-and-configuring-kdump_system-design-guide#configuring-the-kdump-target_configuring-kdump-on-the-command-line[here] for further information on configuring the dump target.
+endif::[]
+
+ifdef::openshift-enterprise[]
+.  Because `kdump` has trouble finding the correct bootimage location on {op-system}, the `KDUMP_BOOTDIR` variable must be manually set in `/etc/kdump.conf`. You can use `/proc/cmdline` to figure out the `ostree` boot location. For example:
++
+[source, terminal]
+----
+BOOT_LOC=/boot$(cat /proc/cmdline | egrep -o "/ostree/.*/vmlinuz" | sed -e "s|/vmlinuz||g")
+sudo sed -i "s|^#KDUMP_BOOTDIR=\"/boot\"|KDUMP_BOOTDIR=\"${BOOT_LOC}\"|" /etc/sysconfig/kdump
+----
+
+.  Configure `/etc/sysconfig/kdump` to not use the default link:https://man7.org/linux/man-pages/man2/kexec_file_load.2.html[file-based `kexec` syscall] for loading the crash kernel.
++
+[source, terminal]
+----
+sudo sed -i "s|^KEXEC_ARGS=\"-s\"|KEXEC_ARGS=\"\"|" /etc/sysconfig/kdump
+----
+endif::[]
+
+.  Enable the `kdump` systemd service.
++
+[source, terminal]
+----
+sudo systemctl enable kdump.service
+----
+
+.  Reboot your system.
++
+[source, terminal]
+----
+sudo systemctl reboot
+----
+
+.  Ensure that `kdump` has loaded a crash kernel by checking that `kdump.service` has started and exited successfully and that `cat /sys/kernel/kexec_crash_loaded` prints `1`.
+
+=== Enabling kdump on day-1
```

Comment:
While it is helpful to show the commands needed to enable `kdump` manually, I think we should also go the extra mile to show what the systemd unit and Machine Config would look like.

Reply:
Hmm it was mentioned that most likely admins would only want to enable kdump on one node or a subset of nodes, but currently machine-specific Machine Configs aren't supported yet. Additionally, we will soon have better FCC sugar in Ignition for doing `kdump`-related configuration. But I'll be happy to add in a MachineConfig example if it would be helpful despite this

Reply:
Ah OK, if the intent is to do something per-node and not roll it out across the cluster, that makes more sense.

We should be clear about the intended use case, then.  Maybe throw a warning at the top of the section with something along the lines of "This should only be enabled per-node in order to debug kernel problems.  It is not recommended to enable kdump on all of your nodes in the cluster"